src/amo/pages/LandingPage/index.js

```diff
@@ -314,11 +318,12 @@ export class LandingPageBase extends React.Component {
   }
 }
 
-export function mapStateToProps(state) {
+export function mapStateToProps(state, ownProps) {
   const { landing, viewContext } = state;
 
   return {
     addonTypeOfResults: landing.addonType,
+    currentURL: getCurrentURL({ state, _config: ownProps._config }),
```

Comment:
Rather than execute this in every `mapStateToProps` (which would happen a lot), how about just passing through the state we need? That would cause the function to only be executed when the state value changes and is more in line with how we typically do things in `mapStateToProps`.

Example:

```diff
diff --git a/src/amo/pages/LandingPage/index.js b/src/amo/pages/LandingPage/index.js
index 67a639446..cceeb9a94 100644
--- a/src/amo/pages/LandingPage/index.js
+++ b/src/amo/pages/LandingPage/index.js
@@ -12,7 +12,7 @@ import { setViewContext } from 'amo/actions/viewContext';
 import LandingAddonsCard from 'amo/components/LandingAddonsCard';
 import NotFound from 'amo/components/ErrorPage/NotFound';
 import Categories from 'amo/components/Categories';
-import { getCurrentURL } from 'amo/utils';
+import { getCanonicalURL } from 'amo/utils';
 import {
   ADDON_TYPE_EXTENSION,
   ADDON_TYPE_THEME,
@@ -212,11 +212,12 @@ export class LandingPageBase extends React.Component {
 
   render() {
     const {
-      currentURL,
+      _config,
       errorHandler,
       featuredAddons,
       highlyRatedAddons,
       loading,
+      locationPathname,
       trendingAddons,
       i18n,
     } = this.props;
@@ -250,7 +251,10 @@ export class LandingPageBase extends React.Component {
       >
         <Helmet>
           <title>{headingText[addonType]}</title>
-          <link rel="canonical" href={currentURL} />
+          <link
+            rel="canonical"
+            href={getCanonicalURL({ pathname: locationPathname, _config })}
+          />
         </Helmet>
 
         {errorHandler.renderErrorIfPresent()}
@@ -319,15 +323,15 @@ export class LandingPageBase extends React.Component {
 }
 
 export function mapStateToProps(state, ownProps) {
-  const { landing, viewContext } = state;
+  const { landing, router, viewContext } = state;
 
   return {
     addonTypeOfResults: landing.addonType,
-    currentURL: getCurrentURL({ state, _config: ownProps._config }),
     context: viewContext.context,
     featuredAddons: landing.featured.results,
     highlyRatedAddons: landing.highlyRated.results,
     loading: landing.loading,
+    locationPathname: router.location.pathname,
     trendingAddons: landing.trending.results,
     resultsLoaded: landing.resultsLoaded && landing.category === null,
   };
diff --git a/src/amo/utils.js b/src/amo/utils.js
index f1f96a1b3..82f0db32e 100644
--- a/src/amo/utils.js
+++ b/src/amo/utils.js
@@ -9,6 +9,7 @@ import NotFound from 'amo/components/ErrorPage/NotFound';
 import ServerError from 'amo/components/ErrorPage/ServerError';
 import { makeQueryString } from 'core/api';
 import type { AppState } from 'amo/store';
+import type { ReactRouterLocationType } from 'core/types/router';
 
 export function getErrorComponent(status: number | null) {
   switch (status) {
@@ -56,6 +57,16 @@ export const makeQueryStringWithUTM = ({
   });
 };
 
+export const getCanonicalURL = ({
+  _config = config,
+  pathname,
+}: {|
+  _config?: typeof config,
+  pathname: string,
+|}): string => {
+  return `${_config.get('baseURL')}${pathname}`;
+};
+
 export const getCurrentURL = ({
   _config = config,
   state,

```

Reply:
> Rather than execute this in every `mapStateToProps` (which would happen a lot), how about just passing through the state we need? That would cause the function to only be executed when the state value changes and is more in line with how we typically do things in `mapStateToProps`.

I think that does make a lot of sense. I knew something was wrong with this patch but could not see what.. Thanks for your input!